File: Assets/Python/Telegram_Factory.py
import protobuf.Telegrams_pb2 as Telegrams
import uuid

class Telegram_Factory:
    def Create_Print_Request(self, message, transaction_id = None):
        request = Telegrams.Request()
        request.command = Telegrams.Request.PRINT
        request.message = message
        return self.Add_Transaction_ID(request, transaction_id)

    def Create_Predict_Request(self, prediction_y, transaction_id = None):
        request = Telegrams.Request()
        request.command = Telegrams.Request.PREDICTION
        for prediction_item in prediction_y:
            request.prediction_y.append(prediction_item)
        return self.Add_Transaction_ID(request, transaction_id)

    def Add_Transaction_ID(self, request, transaction_id):
        if transaction_id == None:
            transaction_id = str(uuid.uuid4())
        request.transaction_id = transaction_id
        return request

File: Assets/TelegramFactory.cs
using System.Collections.Generic;

namespace Assets
{
    public static class TelegramFactory
    {
        public static Telegrams.Request CreatePredictRequest(IEnumerable<float> predict_x)
        {
            var request = new Telegrams.Request();
            request.Command = Telegrams.Request.Types.Command.Predict;
            request.PredictX.AddRange(predict_x);

            return AddTransactionID(request);
        }

        public static Telegrams.Request CreateBeginTrainingRequest()
        {
            var request = new Telegrams.Request();
            request.Command = Telegrams.Request.Types.Command.BeginTraining;

            return AddTransactionID(request);
        }

        public static Telegrams.Request CreateAddTrainingDataRequest(IEnumerable<float> training_x, IEnumerable<float> training_y)
        {
            var request = new Telegrams.Request();
            request.Command = Telegrams.Request.Types.Command.AddTrainingData;
            request.TrainingX.AddRange(training_x);
            request.TrainingY.AddRange(training_y);

            return AddTransactionID(request);
        }

        private static Telegrams.Request AddTransactionID(Telegrams.Request request)
        {
            request.TransactionId = System.Guid.NewGuid().ToString();
            return request;
        }
    }
}
